src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletController.java

```diff
@@ -14,4 +21,20 @@
 public class WalletController {
 
     private final WalletService walletService;
+
+    @GetMapping(value = "/wallets")
+    ResponseEntity<List<WalletDTO>> getWallets() throws AppException {
```

Comment:
Po co to response entity? Całą logikę bym dał do serwisu i w controllerze tylko wywołanie

Reply:
Not sure I understand your comment correctly, You mean that I should change the method signature of service method to return ResponseEntity object?

In my opinion as the ResponseEntity represents an HTTP response (which may consists of response status, message and body), so it is more related to Controller-layer where we prepare and send business output to the view.
It shouldn't be returned by Service-layer where we only handle the business logic. In our case, it is to return a list of all available wallets, which is then used as a body of our http response.

Could you provide an example of your proposed approach of creating a Controller-layer in the projects (Github projects/blog posts etc)?

Reply:
Http response should be in controller layer, agree, but why u use response entity? If wallet is empty just return empty list and status ok, frontend know what to do with it.

Reply:
Given  get method not only handle the situation when empty list is returned